File: src/my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"


class SmartphoneNode : public rclcpp::Node {
    public:
        SmartphoneNode() : rclcpp::Node("smartphone") {
            subscriber = this->
                create_subscription<example_interfaces::msg::String>(
                    "robot_news",
                    10,
                    std::bind(&SmartphoneNode::callbackRobotNews, this, std::placeholders::_1)
                );
            RCLCPP_INFO(this->get_logger(), "Smartphone started");
        }
    
    private:
        void callbackRobotNews(const example_interfaces::msg::String::SharedPtr msg) {
            RCLCPP_INFO(this->get_logger(), "Received: %s", msg->data.c_str());
        }

        rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber;
};


int main(int argc, char **argv) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<SmartphoneNode>());
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/smartphone.py
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String


class SmartphoneNode(Node):
    def __init__(self):
        super().__init__("smartphone")
        self.subscriber = self.create_subscription(String, "robot_news", self.callback_robot_news, 10)
        self.get_logger().info("Smartphone started")

    def callback_robot_news(self, msg: String):
        self.get_logger().info(f"Received data: {msg.data}")


def main(args=None):
    rclpy.init(args=args)
    rclpy.spin(SmartphoneNode())
    rclpy.shutdown()

if __name__ == "__main__":
    main()